package java3;

public class java03 {
public static void main(String[] args)throws Exception {
	//1-1
	//	try {
	//		File file=new File("abc.txt");
	//		if(!file.exists()){
	//			file.createNewFile();
	//		}
	//		PrintWriter pw=new PrintWriter(file);
	//		pw.println("1．NullPointerException:is a RuntimeException. In Java, a special null value can be assigned to an object reference. NullPointerException is thrown when program attempts to use an object reference that has the null value.");
	//		pw.println("2．FileNotFoundException");
	//		pw.println("3．ArrayIndexOutOfBoundsException");
	//		pw.println("4．StringIndexOutOfBoundsException");
	//		pw.println("5．NumberFormatException");
	//		pw.close();
	//		System.out.println("done");
	//	} catch (IOException e) {
	//		e.printStackTrace();
	//	}
	//

	//2-1:
	//	try {
	//		Object object = null;
	//		object.hashCode();
	//	} catch (NullPointerException e) {
	//		System.out.println("スタックトレースも出力する");
	//		e.printStackTrace();
	//	}

	//2-2:
	//		try {
	//			File file=new File("abcd.txt");
	//			Scanner scan=new Scanner(file);
	//		} catch (FileNotFoundException e) {
	//			System.out.println("スタックトレースも出力する");
	//			e.printStackTrace();
	//		}

	//2-3:
//	try {
//		String[] arrayb={"a","b","c"};
//		System.out.println(arrayb[3]);
//	} catch (ArrayIndexOutOfBoundsException e) {
//		System.out.println("スタックトレースも出力する");
//		e.printStackTrace();
//	}

	}
}
